OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs

```diff
@@ -118,14 +123,21 @@ public MissionBrowserLogic(Widget widget, ModData modData, World world, Action o
 						.OrderBy(x => x.Index)
 						.Select(x => x.Preview);
 
-					CreateMissionGroup(kv.Key, previews);
-					allPreviews.AddRange(previews);
+					if (previews.Any())
+					{
+						CreateMissionGroup(kv.Key, previews);
+						allPreviews.AddRange(previews);
+					}
 				}
 			}
 
 			// Add an additional group for loose missions
 			var loosePreviews = modData.MapCache
-				.Where(p => p.Status == MapStatus.Available && p.Visibility.HasFlag(MapVisibility.MissionSelector) && !allPreviews.Any(a => a.Uid == p.Uid));
+				.Where(p => p.Status == MapStatus.Available
+					&& p.Visibility.HasFlag(MapVisibility.MissionSelector)
+					&& !allPreviews.Any(a => a.Uid == p.Uid)
+					&& ((string.IsNullOrEmpty(p.RequiresMissionFinished) || (finishedMissions != null && finishedMissions.Contains(p.RequiresMissionFinished)))
```

Comment:
These checks can then be replaced by having startButton.IsDisabled check an extra flag that is set after the map rules have been loaded (refer to how the mission briefing is loaded) to enable the button only if the prereqs are available.

Reply:
We can optionally show a label somewhere saying that earlier missions must be finished before this can be started.

Reply:
I realize that this approach isn't ideal, but in the future we should be planning to remove the mission list from the side completely, and use a selection of graphical maps for progression instead.